<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>Connectivity Map Data API and Apps</title>

		<meta name="description" content="Connectivity Map Data API and Apps Presentation">
		<meta name="author" content="Corey Flynn">

		<meta name="apple-mobile-web-app-capable" content="yes" />
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<link rel="stylesheet" href="../reveal/css/reveal.min.css">
		<link rel="stylesheet" href="../reveal/css/theme/serif.css" id="theme">
		<link rel="stylesheet" href="http://cmap.github.io/barista/barista.main.min.css">

		<!-- For syntax highlighting -->
		<link rel="stylesheet" href="../reveal/lib/css/zenburn.css">

		<!-- If the query includes 'print-pdf', use the PDF print sheet -->
		<script>
			document.write( '<link rel="stylesheet" href="../reveal/css/print/' + ( window.location.search.match( /print-pdf/gi ) ? 'pdf' : 'paper' ) + '.css" type="text/css" media="print">' );
		</script>

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

		<div class="reveal">

			<!-- Any section element inside of this container is displayed as a slide -->
			<div class="slides">
				<!-- preliminaries -->
				<section>
					<!-- title -->
					<section>
						<h1>Connectivity Map Data and Tools</h1>
						<div style="min-height:100px"></div>
						<h1><font color="red">Hands On</font></h1>
					</section>

					<!-- who am I? -->
					<section>
						<h2>Corey Flynn</h2>
				        <p><a href="mailto:cflynn@broadinstitute.org" style="width:1000; text-align:center"> cflynn@broadinstitute.org</a></p>
				        <p> Bioinformatics Scientist II</p>
				        <p> Connectivity Map,Golub Lab</p>
				        <p> Cancer Program</p>
					</section>

					<!-- who is Rajiv? -->
					<section>
						<h2>Rajiv Narayan</h2>
				        <p><a href="mailto:narayan@broadinstitute.org" style="width:1000; text-align:center"> narayan@broadinstitute.org</a></p>
				        <p>Computational Biologist</p>
				        <p> Connectivity Map,Golub Lab</p>
				        <p> Cancer Program</p>
					</section>
				</section>

				<!-- follow these slides -->
				<section>
					<p> follow these slides at </p>
			        <a href="http://coreyflynn.github.io/SoapBox/CMAP_hands_on">http://coreyflynn.github.io/SoapBox/CMAP_hands_on</a>
				</section>
				
				<!-- LINCS Overview -->
				<section>
					<section>
						<h2>LINCS and CMap Overview</h2>
					</section>

					<!-- LINCS Data Cube -->
					<section>
						<h2>LINCS Data Concept</h2>
				        <img src="img/LINCSCube.png" style="border:none; box-shadow: 0 0 0 0;">
				        <ul>
							<li>perturbations scalable to genome</li>
							<li>high information content read-outs  (e.g. gene expression)</li>
							<li>inexpensive</li>
							<li>mechanism to query database</li>
				        </ul>
					</section>

					<!-- LINCS Goals -->
					<section>
						<h2>GOAL: A look-up table for functional annoation</h2>
						<div style="min-height:50px"></div>
						<ul>
							<li style="color:red;">Given a gene or protein
								<ul style="color:black;">
									<li>What other proteins have similar cellular effects?</li>
									<li>What compounds impinge upon its activity?</li>
								</ul>
							</li>
						</ul>
					</section>

					<!-- LINCS Goals -->
					<section>
						<h2>GOAL: A look-up table for functional annoation</h2>
						<div style="min-height:50px"></div>
						<ul>
							<li style="color:red;">Given a small molecule compound
								<ul style="color:black;">
									<li>What is its mechanism of action?</li>
									<li>What are all of its effects on cells?</li>
									<li>Is it unique?</li>
								</ul>
							</li>
						</ul>
					</section>

					<!-- LINCS Goals -->
					<section>
						<h2>GOAL: A look-up table for functional annoation</h2>
						<div style="min-height:50px"></div>
						<ul>
							<li style="color:red;">Given a biological state
								<ul style="color:black;">
									<li>What protein activities relate to that state?</li>
									<li>What compound activities relate to that state?</li>
								</ul>
							</li>
						</ul>
					</section>

					<!-- expirimental flow -->
					<section>
						<img src="img/experimental_flow@2x.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
					</section>

					<!-- L1000 comparison matrix -->
					<section>
						<h2>L1000 expression profiling</h2>
						<img src="img/L100_comparison_matrix.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
					</section>

					<!-- Magnitude of LINCS -->
					<section>
						<h2>LINCS Expansion</h2>
						<div class="cmap-spacer-large"></div>
						<div id="LINCS_bar_target"></div>
					</section>

					<!-- current LINCS Dataset -->
					<section>
						<h2>Current LINCS CMap Dataset</h2>
						<iframe src="http://apps.lincscloud.org/data_synopsis/" frameborder="0" width="1000" height="520" seamless></iframe>
					</section>
				</section>

				<!-- Data Normalization -->
				<section>
					<section>
						<h2>Data Collection</h2>
					</section>

					<!-- Scan data -->
					<section>
						<h2>Scan data</h2>
						<div style="min-height: 50px"></div>
						<img src="img/LINCS_img/plates.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Cell lysates are scanned in 384 well plates in the L1000 assay</p>
					</section>

					<!-- Deconvolute -->
					<section>
						<h2>Bead deconvolution</h2>
						<div style="min-height: 50px"></div>
						<img src="img/LINCS_img/deconvolute.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>1000 gene measurements are decovoluted from 500 bead colors</p>
					</section>

					<!-- Normalization -->
					<section>
						<h2>Normalization</h2>
						<div style="min-height: 50px"></div>
						<img src="img/LINCS_img/normalization.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Scale profiles based on expected levels of invariant gene sets and adjust to expected distribution</p>
					</section>

					<!-- Inference -->
					<section>
						<h2>Inference</h2>
						<div style="min-height: 50px"></div>
						<img src="img/LINCS_img/inference.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Use regression models to impute the rest of the transcriptome</p>
					</section>

					<!-- Z-score -->
					<section>
						<h2>Z-score</h2>
						<div style="min-height: 50px"></div>
						<img src="img/LINCS_img/z_score.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Compute Z-scores to determine differential expression</p>
					</section>

					<!-- Differential Expression -->
					<section>
						<h2>Differential Expression</h2>
						<div style="min-height: 50px"></div>
						<img src="img/LINCS_img/collapse.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Average between replicates weighted by agreement</p>
					</section>

					<!-- CGS -->
					<section>
						<h2>Consensus Gene Signatures</h2>
						<div style="min-height: 50px"></div>
						<img src="img/LINCS_img/CGS.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Integrate across knockdown signatures targeting the same gene to generate consensus gene signatures</p>
					</section>

					<!-- Assembled matrix -->
					<section>
						<h2>Data Matrix</h2>
						<div style="min-height: 50px"></div>
						<img src="img/sig_tools/sig_matrix.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Assemble signatures into a unified matrix</p>
					</section>

				</section>

				<!-- data resources -->
				<section>
					<section>
						<h2>Data Resources</h2>
						<div style="min-height: 50px"></div>
						<img src="img/DataResources.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
					</section>

					<!-- Apps -->
					<section>
						<div style="min-height: 50px"></div>
						<img src="img/DataResourcesApps.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
					</section>

					<!-- API -->
					<section>
						<div style="min-height: 50px"></div>
						<img src="img/DataResourcesAPI.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
					</section>					

					<!-- C3 -->
					<section>
						<div style="min-height: 50px"></div>
						<img src="img/DataResourcesC3.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
					</section>
				</section>

				<!-- Tools -->
				<section>
					<section>
						<h2>Sig Tools</h2>
					</section>

					<!-- Sig Query -->
					<section>
						<h2>Query Tool</h2>
						<div style="min-height: 50px"></div>
						<img src="img/sig_tools/sig_query.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Compare one signature to all other signatures in the database looking for connections</p>
					</section>

					<!-- Sig Summly -->
					<section>
						<h2>Summly Tool</h2>
						<div style="min-height: 50px"></div>
						<img src="img/sig_tools/sig_summly.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Summarize the results of a query to focus on the most reliable connections</p>
					</section>

					<!-- Sig Introspect -->
					<section>
						<h2>Introspect Tool</h2>
						<div style="min-height: 50px"></div>
						<img src="img/sig_tools/sig_introspect.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Look for consistency in related signatures</p>
					</section>

					<!-- Other Tools: Info-->
					<section>
						<h2>Sig Info</h2>
						<div style="min-height: 50px"></div>
						<img src="img/sig_tools/sig_info.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Get metadata for signatures</p>
					</section>

					<!-- Other Tools: Slice -->
					<section>
						<h2>Sig Slice</h2>
						<div style="min-height: 50px"></div>
						<img src="img/sig_tools/sig_slice.png" alt="" style="border:none; box-shadow: 0 0 0 0;">
						<p>Make subsets of the data matrix</p>
					</section>

				</section>

			</div>

		</div>

		<script src="../reveal/lib/js/head.min.js"></script>
		<script src="../reveal/js/reveal.min.js"></script>
		<script src="http://cmap.github.io/barista/barista.main.min.js"></script>

		<script>

			// Full list of configuration options available here:
			// https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				controls: true,
				progress: true,
				history: true,
				center: true,

				theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
				transition: Reveal.getQueryHash().transition || 'default', // default/cube/page/concave/zoom/linear/fade/none

				// Optional libraries used to extend on reveal.js
				dependencies: [
					{ src: '../reveal/lib/js/classList.js', condition: function() { return !document.body.classList; } },
					{ src: '../reveal/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: '../reveal/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: '../reveal/plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
					{ src: '../reveal/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
					{ src: '../reveal/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }
				]
			});

		</script>
		<script type="text/javascript">
			// data model
			bar_model = new Barista.Models.BarPlotModel({data: [520,7200],
														 title: 'Number of profiles'});

			// view
			bar_view = new Barista.Views.BarPlotView({el: $('#LINCS_bar_target'),
													  model: bar_model,
													  plot_height:300,
													  min_lock: 0,
													  margin: 60,
													  fg_color: "red",
													});
			$('#LINCS_bar_target').click(function(){
				if (bar_model.get('data').length == 1){
					bar_model.set({data: [520,7200]});
					bar_model.set({data_labels: ['Pilot','Pilot Extension']});
					return
				}
				if (bar_model.get('data').length == 2){
					bar_model.set({data: [520, 7200,1328098]});
					bar_model.set({data_labels: ['Pilot','Pilot Extension','LINCS']});
					return
				}else{
					bar_model.set({data: [520]});
					bar_model.set({data_labels: ['Pilot']});
					return
				}
			});
			bar_model.set({data: [520]});
			bar_model.set({data_labels: ['Pilot']});
			$('.controls_layer').hide();

		</script>

	</body>
</html>
